src/tasks/tasks.proj

```diff
@@ -41,7 +41,9 @@
   <Target Name="GetTasksSrc"
           DependsOnTargets="PrepareProjectReferences">
     <PropertyGroup>
-      <TasksIntermediateFile>$([MSBuild]::NormalizePath('$(ArtifactsObjDir)', '$(MSBuildProjectName)', 'Debug', 'build-semaphore.txt'))</TasksIntermediateFile>
+      <TasksIntermediateFile Condition="'$(TargetOS)' == '' or '$(TargetArchitecture)' == ''">$([MSBuild]::NormalizePath('$(ArtifactsObjDir)', '$(MSBuildProjectName)', 'Debug', 'build-semaphore.txt'))</TasksIntermediateFile>
```

Comment:
This project ultimately imports Directory.Build.props on the root where TargetArchitecture/TargetOS are set for all cases, which means this condition cannot be false (even if we build just this project e.g. `./dotnet.sh msbuild src/tasks/tasks.proj`).